gprofiler/profilers/java.py

```diff
@@ -333,21 +405,51 @@ def __init__(
         java_version_check: bool,
         java_async_profiler_mode: str,
         java_async_profiler_safemode: int,
+        java_safemode: bool,
         java_mode: str,
     ):
         assert java_mode == "ap", "Java profiler should not be initialized, wrong java_mode value given"
         super().__init__(frequency, duration, stop_event, storage_dir)
 
+        if java_safemode:
+            assert java_version_check, "Java version checks are mandatory in --java-safemode"
+            assert java_async_profiler_safemode == 127, "Async-profiler safemode must be set to 127 in --java-safemode"
+
         # async-profiler accepts interval between samples (nanoseconds)
         self._interval = int((1 / frequency) * 1000_000_000)
         self._buildids = java_async_profiler_buildids
         self._version_check = java_version_check
+        if not java_version_check:
+            logger.warning("Java version checks are disabled")
         self._mode = java_async_profiler_mode
         self._safemode = java_async_profiler_safemode
         self._saved_mlock: Optional[int] = None
 
     def _is_jdk_version_supported(self, java_version_cmd_output: str) -> bool:
-        return all(exclusion not in java_version_cmd_output for exclusion in self.JDK_EXCLUSIONS)
+        # Failsafe
+        for exclusion in self.JDK_EXCLUSIONS:
+            if exclusion in java_version_cmd_output:
+                logger.error(f"Spotted unsupported keyword {exclusion} in java version, not profiling")
+                return False
+
+        try:
+            jvm_version = parse_jvm_version(java_version_cmd_output)
+            logger.info(f"Checking support for java version {jvm_version}")
+        except Exception as e:
+            logger.error(f"Failed to parse java -version output {java_version_cmd_output}: {e}")
+            return False
```

Comment:
This part (and generally - failing the `JavaProfiler` if the JDK version is too old) - should only be enabled if `java_safemode` is set.